proto/cacheclient.proto

```diff
@@ -391,3 +395,40 @@ message _ListLengthResponse {
     _Missing missing = 2;
   }
 }
+
+message _IntegerGetRequest {
+  bytes integer_name = 1;
+}
+
+message _IntegerGetResponse {
+  message _Found {
+    int64 value = 1;
+  }
+
+  message _Missing {}
+
+  oneof integer {
+    _Found found = 1;
+    _Missing missing = 2;
+  }
+}
+
+message _IntegerIncrementRequest {
+  bytes integer_name = 1;
+  int64 increment_by = 2;
```

Comment:
> If increment is called without a provided value to increment by, it will default to incrementing by 1.

This will cause a weirdness where if someone does not do a special-case check for `0` we will increment by 1. They might truly want to call increment(0) because of details of their software. Remember, "unset" in `proto3` syntax for the int64 type is the nonnull value `0`.

Reply:
Maybe it is safest to require a value for increment_by ?

Reply:
It is not possible to not provide a value for increment_by in `proto3`. A missing value on the serialized bits sent over the network is equivalent to setting `0`. https://developers.google.com/protocol-buffers/docs/proto3#default and some implementations may in fact elide the bits for this field when `0` was explicitly set.

You could use `oneof` and define messages like `message Increment {}` `message Add { int64 amount = 1; }` to explicitly choose the intended semantic

Reply:
> Remember, "unset" in `proto3` syntax for the int64 type is the nonnull value `0`.

TIL. I did not know this. Agreed that this means having default if no value is provided be 1 does not make sense. We could have that be the case in the SDKs if we want, but agreed it should not be the case from server's perspective.

Reply:
I am supportive of letting the number be what it is and doing "add" with the provided number on the service. I agree that the SDK can form the message however it likes.